plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2958,11 +2985,27 @@ private KVMPhysicalDisk getPhysicalDiskPrimaryStore(PrimaryDataStoreTO primaryDa
     protected void setDiskIoDriver(DiskDef disk) {
         if (getHypervisorLibvirtVersion() >= HYPERVISOR_LIBVIRT_VERSION_SUPPORTS_IO_URING
                 && getHypervisorQemuVersion() >= HYPERVISOR_QEMU_VERSION_SUPPORTS_IO_URING
-                && AgentPropertiesFileHandler.getPropertyValue(AgentProperties.ENABLE_IO_URING)) {
+                && isIoUringEnabled()) {
             disk.setIoDriver(DiskDef.IoDriver.IOURING);
         }
     }
 
+    /**
+     * IO_URING supported if the property 'enable.io.uring' is set to true OR it is supported by qemu
+     */
+    private boolean isIoUringEnabled() {
+        Boolean propertyValue = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.ENABLE_IO_URING);
+        return BooleanUtils.isTrue(propertyValue) || isBaseOsUbuntu() || isIoUringSupportedByQemu();
```

Comment:
In my opiniom, the logic is
If enable.io.uring is true, enable io uring.
If enable.io.uring is false, disable io uring.
If it is not set, determine by ldd

Reply:
Thanks @weizhouapache, I have applied this - only difference I added: when it is not set then enable for ubuntu or determine ldd as per @wido @GabrielBrascher comments this wasn't an issue for Ubuntu (also this check is performed when meeting the supported qemu and libvirt versions)